include/pybind11/cast.h

```diff
@@ -207,6 +207,19 @@ PYBIND11_NOINLINE inline handle get_type_handle(const std::type_info &tp, bool t
     return handle(type_info ? ((PyObject *) type_info->type) : nullptr);
 }
 
+// Searches all_type_info for a registered instance, i.e. moving up the inheritance hierarchy.
```

Comment:
> i.e. moving up the inheritance hierarchy.

Is that comment correct? I don't believe so. Where is this thing climbing the hierarchy?

According to `all_type_info`, it's doing something like that, but not really the full hierarchy:

```
/**
 * Extracts vector of type_info pointers of pybind-registered roots of the given Python type.  Will
 * be just 1 pybind type for the Python type of a pybind-registered class, or for any Python-side
 * derived class that uses single inheritance.  Will contain as many types as required for a Python
 * class that uses multiple inheritance to inherit (directly or indirectly) from multiple
 * pybind-registered classes.  Will be empty if neither the type nor any base classes are
 * pybind-registered.
 *
 * The value is cached for the lifetime of the Python type.
 */
```

So `all_type_info` seems to just be finding the _first_ pybind11 class(es) up the hierarchy, but I don't see the entries in `all_type_info` ordered by inheritance?

Reply:
I reworded the comment for clarity.
I didn't mean to claim it's searching the entire hierarchy, I just wanted to point to `all_type_info`. I added `()` to make that more obvious, in addition to shuffling the words.
I also removed the word `up`, although I think it was correct, but it doesn't really need to be mentioned here, because `all_type_info` has a pretty detailed explanation.